### DISCUSSION TITLE
Support postcss.config.cjs

### DISCUSSION THREAD

**Discussion by JohnDaly** [vercel/next.js#38497] [Ideas]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

    Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 21.5.0: Tue Apr 26 21:08:37 PDT 2022; root:xnu-8020.121.3~4/RELEASE_ARM64_T6000
    Binaries:
      Node: 16.14.0
      npm: 8.3.1
      Yarn: 1.22.17
      pnpm: 6.11.0
    Relevant packages:
      next: 12.2.1
      eslint-config-next: 12.2.1
      react: 18.2.0
      react-dom: 18.2.0

### Describe the Bug

When Next attempts to find a PostCSS configuration, it doesn't look for files ending in `.config.cjs`, and the configuration is not loaded.

### Expected Behavior

PostCSS supports using `postcss.config.cjs` as a configuration file (https://github.com/postcss/postcss/issues/1329#issuecomment-877863139). Next should find and load this configuration file.

### Link to reproduction

https://stackblitz.com/edit/vercel-next-js-dxk18b?file=pages/index.tsx

### To Reproduce

1. Set up a Next.js app with Tailwind by following the instructions at: https://tailwindcss.com/docs/guides/nextjs
2. Create a component with Tailwind classnames
3. Run the app and see that the Tailwind styles are applied to the component
4. Add `"type": "module"` to the `package.json` of the Next.js app
5. Update `next.config.js`, `postcss.config.js`, and `tailwind.config.js` to use `.config.cjs` file extension
6. Run the app and see that the Tailwind styles are no longer applied to the component

**Comment by weyert** [comment]
I am having the same issue and blocking me from switching to ESM project when using Tailwindcss

**Comment by damianobarbati** [comment]
Just stumbled on this, any news about it?

**Comment by jeiea** [comment]
It is supported, I just misconfigured tailwind.config.cjs by renaming component directory.

**Reply by CMCDragonkai** [reply]
How did you do this?
